src/coverlet.core/Coverage.cs

```diff
@@ -170,11 +183,22 @@ public CoverageResult GetCoverageResult()
                 InstrumentationHelper.RestoreOriginalModule(result.ModulePath, _identifier);
             }
 
-            var coverageResult = new CoverageResult { Identifier = _identifier, Modules = modules, InstrumentedResults = _results };
+            var coverageResult = new CoverageResult
+                {Identifier = _identifier, Modules = modules, InstrumentedResults = _results};
 
             if (!string.IsNullOrEmpty(_mergeWith) && !string.IsNullOrWhiteSpace(_mergeWith) && File.Exists(_mergeWith))
             {
-                string json = File.ReadAllText(_mergeWith);
+                var json = RetryHelper.Do(() =>
+                {
+                    using (var file = File.Open(_mergeWith, FileMode.Open, FileAccess.Read, FileShare.None))
+                    {
+                        using (var reader = new StreamReader(file))
+                        {
+                            return reader.ReadToEnd();
+                        }
+                    }
+                }, () => TimeSpan.FromMilliseconds(100), 5);
```

Comment:
I'm not sure about max attempt, I would try for "more standard seconds" 10 or 30.